pennylane/__init__.py

```diff
@@ -227,13 +230,21 @@ def circuit():
     raise DeviceError("Device does not exist. Make sure the required plugin is installed.")
 
 
-def grad(func, argnum=None):
+make_vjp = unary_to_nary(_make_vjp)
+
+
+class grad:
     """Returns the gradient as a callable function of (functions of) QNodes.
 
     Function arguments with the property ``requires_grad`` set to ``False``
     will automatically be excluded from the gradient computation, unless
     the ``argnum`` keyword argument is passed.
 
+    When the output gradient function is executed, both the forward pass
+    *and* the backward pass will be performed in order to
+    compute the gradient. The value of the forward pass is available via the
+    :attr:`~.forward` property.
+
     Args:
         func (function): a plain QNode, or a Python function that contains
             a combination of quantum and classical nodes
```

Comment:
Do we still need to refer to `argnum` as a keyword argument below? `qml.grad(cost_fn, 0)` seems to work.

Reply:
Yeah, I'm not sure why this was done. I can move it.